services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/controller/SubnetController.java

```diff
@@ -86,59 +101,106 @@ public SubnetStateJson createSubnetState(@PathVariable String projectid, @Reques
             RestPreconditionsUtil.verifyResourceFound(inSubnetState.getVpcId());
             RestPreconditionsUtil.populateResourceProjectId(inSubnetState, projectid);
 
-            this.subnetDatabaseService.addSubnet(inSubnetState);
-
-            subnetState = this.subnetDatabaseService.getBySubnetId(inSubnetState.getId());
-            if (subnetState == null) {
-                throw new ResourcePersistenceException();
-            }
+            //Allocate Gateway Mac
+            CompletableFuture<MacStateJson> macFuture = CompletableFuture.supplyAsync(() -> {
+                try {
+                    return this.subnetService.allocateMacGateway(projectid, inSubnetState.getVpcId(), portId);
+                } catch (Exception e) {
+                    throw new CompletionException(e);
+                }
+            }, ThreadPoolExecutorUtils.SELECT_POOL_EXECUTOR).handle((s, e) -> {
+                macResponseAtomic.set(s);
```

Comment:
This is the way to execute asynchronously， because we can not use MacStateJson directly here (I mean it is in  CompletableFuture.handle() method).
So I use the atomic reference